internal/exec/stages/files/files.go

```diff
@@ -182,6 +182,11 @@ func (tmp dirEntry) create(l *log.Logger, u util.Util) error {
 				return err
 			}
 		}
+
+		if err = os.Chmod(path, os.FileMode(d.Mode)); err != nil {
```

Comment:
This code, as well as the corresponding code in `fileEntry.create`, sets the mode but not the ownership.  That inconsistency seems likely to confuse someone.